### PR TITLE
Allow configuring techdocs

### PR DESCRIPTION
**Pull request by iain-b**

### CODE REVIEW

File: backstage/templates/backstage-app-config.yaml [hunk 1]
```diff
@@ -22,8 +22,6 @@ data:
   {{- end }}
   APP_CONFIG_backend_baseUrl: {{ .Values.appConfig.backend.baseUrl | quote | quote }}
   APP_CONFIG_backend_cors_origin: {{ .Values.appConfig.backend.cors.origin | quote | quote }}
-  APP_CONFIG_techdocs_storageUrl: {{ .Values.appConfig.techdocs.storageUrl | quote | quote }}
```
Comment (iain-b) [inline backstage/templates/backstage-app-config.yaml:25]: These are deprecated upstream and I don't think they're needed. I'm open to correction but AFAIK the default uses the discovery api to find the techdocs static/docs endpoint which uses the publisher config to read the files.

File: backstage/values.yaml [hunk 2]
```diff
@@ -152,8 +152,8 @@ appConfig:
   sentry:
     organization: spotify
   techdocs:
-    storageUrl: https://demo.example.com/api/techdocs/static/docs
-    requestUrl: https://demo.example.com/api/techdocs
+    generators:
```
Comment (iain-b) [inline backstage/values.yaml:155]: We won't need to specify credentials via app config to access s3 (AFAIK Jussi is configuring policies for the backend pod role?) but for other publishers secrets would need to be set in the `backstage-values` secret which I believe is manually created. (Assuming we need to support other publishers & alternatively/additionally with BYO storage we could use external secrets)

Comment (Xantier) [reply backstage/values.yaml:155]: No need for creds, they'll be pulled from the instance for reading

Comment (Xantier) [reply backstage/values.yaml:155]: Publishing would be handled outside of backstage at least for contentful (and ideally for everyone else as well)

**Review by Xantier — APPROVED** [review]

File: backstage/values.yaml (hunk 2 above)
Comment (iain-b) [reply backstage/values.yaml:155]: It's a little confusing semantically but the publisher config is also used in the read-only path (builder: 'external'). So if we wanted to say read from GCS we'd need to configure the secret in the publisher config. _I think_. But this is just an aside that I think we could support that if / when we came to it.

Comment (Xantier) [reply backstage/values.yaml:155]: Yup, it uses the same config but all AWS, GCP and Azure should be able to be set up with env variables so we don't need to add anything on this side I think

**Pull request merged**